.github/workflows/docs.yml

```diff
@@ -1,29 +1,246 @@
-  pull_request:
-    types:
-      - opened
-      - synchronize
-      - reopened
-      - ready_for_review
-  # Enable manual trigger for easy debugging
-  workflow_dispatch:
-
-permissions:
-  contents: read
-  # Optional: allow read access to pull request. Use with `only-new-issues` option.
-  # pull-requests: read
+                set -o pipefail
+                # Fallback to /usr/local for backwards compatability
+                prefix_dir="${RUNNER_TEMP:-/usr/local}"
 
-jobs:
-  lint:
-    runs-on: "${{ github.repository == 'dagger/dagger' && 'dagger-g2-v0-12-5-4c' || 'ubuntu-latest' }}"
-    timeout-minutes: 10
-    steps:
-      - uses: actions/checkout@v4
-      - name: "docs lint"
-        uses: ./.github/actions/call
-        with:
-          function: "docs lint"
+                # Ensure the dir is writable otherwise fallback to tmpdir
+                if [[ ! -d "$prefix_dir" ]] || [[ ! -w "$prefix_dir" ]]; then
+                    prefix_dir="$(mktemp -d)"
+                fi
+                printf '%s/bin' "$prefix_dir" >> $GITHUB_PATH
+
+                # If the dagger version is 'latest', set the version back to an empty
+                # string. This allows the install script to detect and install the latest
+                # version itself
+                if [[ "$DAGGER_VERSION" == "latest" ]]; then
+                  DAGGER_VERSION=
+                fi
+
+                # The install.sh script creates path ${prefix_dir}/bin
+                curl -fsS https://dl.dagger.io/dagger/install.sh | BIN_DIR=${prefix_dir}/bin sh
+              env:
+                DAGGER_VERSION: v0.12.5
+              shell: bash
+            - name: scripts/warm-engine.sh
+              id: warm-engine
+              run: |
+                #!/bin/bash
+
+                # Make sure not to load any implicit module
+                cd $(mktemp -d)
+                # Run a simple query to "warm up" the engine
+                echo '{directory{id}}' | dagger query
+              shell: bash
+            - name: scripts/exec.sh
+              id: exec
+              run: |
+                #!/bin/bash --noprofile --norc -e -o pipefail
+
+                if [[ -n "$DEBUG" && "$DEBUG" != "0" ]]; then
+                    set -x
+                    env
+                    which dagger
+                    pwd
+                    ls -l
+                    ps aux
+                fi
+
+                # Detect if a dev engine is available, if so: use that
+                # We don't rely on PATH because the GHA runner messes with that
+                if [[ -n "$_EXPERIMENTAL_DAGGER_CLI_BIN" ]]; then
+                    export PATH=$(dirname "$_EXPERIMENTAL_DAGGER_CLI_BIN"):$PATH
+                fi
+
+                GITHUB_OUTPUT="${GITHUB_OUTPUT:=github-output.txt}"
+                GITHUB_STEP_SUMMARY="${GITHUB_STEP_SUMMARY:=github-summary.md}"
+                export NO_COLOR="${NO_COLOR:=1}" # Disable colors in dagger logs
```

Comment:
Color? :pleading_face:

Reply:
I needed to disable color to avoid polluting my markdown report - did you see that feature?

I realize now it's a tradeoff between 1) neat logs in markdown report, and 2) colors in GHA logs

Reply:
Mm, I see. Yeah, that's alright for now I think.

Here's an idea: we could use something like `ansi2txt` from https://github.com/kilobyte/colorized-logs to take the colored output and remove it for the markdown report.

(I can't see where that feature is btw, might be missing something).

Reply:
Ah, I found it. FYI, there's still some control color codes in there sometimes: https://github.com/dagger/dagger/actions/runs/10648614702?pr=8241

Reply:
Could it be because of nested dagger engines?